src/table.rs

```diff
@@ -60,9 +81,42 @@ impl Table {
         Ok(Self { schema, batches })
     }
 
+    /// Creates a new table from a schema, a vector of record batches, and a chunked geometry array.
+    ///
+    /// Returns an error if a record batch's schema fields do not match the
+    /// top-level schema's fields, or if the batches are empty.
```

Comment:
As a general statement I like the format

```
# Errors

If a record batch's schema fields do not match the
top-level schema's fields, or if the batches are empty.
```

because it's easy to quickly parse what the section is about from the big header